Sources/xcproj/PBXProj.swift

```diff
@@ -3,8 +3,133 @@ import PathKit
 
 /// Represents a .pbxproj file
 public class PBXProj: Decodable {
+    public class PBXProjObjects: Equatable {
+        // MARK: - Properties
+        public var buildFiles: ReferenceableCache<PBXBuildFile> = ReferenceableCache<PBXBuildFile>()
+        public var aggregateTargets: ReferenceableCache<PBXAggregateTarget> = ReferenceableCache<PBXAggregateTarget>()
+        public var containerItemProxies: ReferenceableCache<PBXContainerItemProxy> = ReferenceableCache<PBXContainerItemProxy>()
+        public var copyFilesBuildPhases: ReferenceableCache<PBXCopyFilesBuildPhase> = ReferenceableCache<PBXCopyFilesBuildPhase>()
+        public var groups: ReferenceableCache<PBXGroup> = ReferenceableCache<PBXGroup>()
+        public var fileElements: ReferenceableCache<PBXFileElement> = ReferenceableCache<PBXFileElement>()
+        public var configurationLists: ReferenceableCache<XCConfigurationList> = ReferenceableCache<XCConfigurationList>()
+        public var versionGroups: ReferenceableCache<XCVersionGroup> = ReferenceableCache<XCVersionGroup>()
+        public var buildConfigurations: ReferenceableCache<XCBuildConfiguration> = ReferenceableCache<XCBuildConfiguration>()
+        public var variantGroups: ReferenceableCache<PBXVariantGroup> = ReferenceableCache<PBXVariantGroup>()
+        public var targetDependencies: ReferenceableCache<PBXTargetDependency> = ReferenceableCache<PBXTargetDependency>()
+        public var sourcesBuildPhases: ReferenceableCache<PBXSourcesBuildPhase> = ReferenceableCache<PBXSourcesBuildPhase>()
+        public var shellScriptBuildPhases: ReferenceableCache<PBXShellScriptBuildPhase> = ReferenceableCache<PBXShellScriptBuildPhase>()
+        public var resourcesBuildPhases: ReferenceableCache<PBXResourcesBuildPhase> = ReferenceableCache<PBXResourcesBuildPhase>()
+        public var frameworksBuildPhases: ReferenceableCache<PBXFrameworksBuildPhase> = ReferenceableCache<PBXFrameworksBuildPhase>()
+        public var headersBuildPhases: ReferenceableCache<PBXHeadersBuildPhase> = ReferenceableCache<PBXHeadersBuildPhase>()
+        public var nativeTargets: ReferenceableCache<PBXNativeTarget> = ReferenceableCache<PBXNativeTarget>()
+        public var fileReferences: ReferenceableCache<PBXFileReference> = ReferenceableCache<PBXFileReference>()
+        public var projects: ReferenceableCache<PBXProject> = ReferenceableCache<PBXProject>()
+        public var referenceProxies: ReferenceableCache<PBXReferenceProxy> = ReferenceableCache<PBXReferenceProxy>()
+
+        // Open Questions about buildPhases
+        // - Unsure if there is value in making this a dictionary
+        // - Will the ordering of the phases be unstable given we are retrieving values from a dictionary
+        // and this might require a sort of some kind (compare references strings?)
+        // - Should we convert the build phases objects back to arrays
```

Comment:
For what it's worth, I don't use this property from XcodeGen. I can't see how the order of these values could matter. It's not the order they are defined in the project. That order is defined on each target. 
I think this could be typed as [String: PBXBuildPhase]

Reply:
Yeah, unless I miss something I don't think the order matters since the order is specified in the target that references them. I think we can keep them as dictionaries.
I use the property from `xclint` but it seems that it's very concrete to the library use case. Feel free to deprecate it since I can extend the `PBXProj` from there.

Reply:
Ok I'll change these to dictionaries. I had a test failure in XCodeGen that I assumed was due to ordering but turned out the test just needed to be updated.